libraries/chain/event_evaluator.cpp

```diff
@@ -50,6 +79,10 @@ void_result event_create_evaluator::do_evaluate(const event_create_operation& op
    event_group_id = resolved_event_group_id;
    //const event_group_object& event_group = event_group_id(d);
 
+   FC_ASSERT(trx_state->_is_proposed_trx
```

Comment:
We can remove the entire test/assertion, IMHO. impact is minimal since witness account is only affects and they are multisig so it's easier for them to use a proposal anyways